src/ir/comp.rs

```diff
@@ -481,6 +486,10 @@ impl CompInfo {
         debug!("CompInfo::from_ty({:?}, {:?})", kind, cursor);
 
         let mut ci = CompInfo::new(kind);
+        ci.is_forward_declaration = location.map_or(true, |cur|
```

Comment:
we could write this as:

```rust
ci.is_forward_declaration = location.map_or(true, |cur| {
    match cur.kind() {
        CXCursor_StructDecl |
        CXCursor_UnionDecl |
        CXCursor_ClassDecl => !cur.is_definition(),
        _ => false,
    }
});
```

Which I think it's a bit more legible, what do you think?